astropy/table/column.py

```diff
@@ -111,8 +88,8 @@ def __setslice__(self, start, stop, val):
                              .format(self.__class__.__name__))
 
 
-class ColumnInfo(DataInfo):
-    attrs_from_parent = DataInfo.attr_names
+class ColumnInfo(BaseColumnInfo):
```

Comment:
I know this might be a bit of overdesign, but I want to have an info class for `BaseColumn` objects that is a separate branch in the inheritance tree.  Basically this to leave room for a subclass of `ColumnInfo`.  For instance, imagine one day it is useful to have a `MaskedColumnInfo`, and you want to test for `has_info_class(ColumnInfo)` (either normal or masked).  If you just have `BaseColumnInfo` then that pulls  in `MixinInfo`.

One thing I was imagining is using the `info` class everywhere to infer the behavior of the object instead of the class of the object itself.  In other words replace `isinstance(col, BaseColumn)` with `has_info_class(ColumnInfo)` everywhere.  Again at this moment this probably falls in the category of overdesign, but now with `info` we do have the freedom to move there as additional use cases come up.  Certainly for mixins this is crucial.